client/src/main/java/io/pravega/client/stream/impl/LargeEventWriter.java

```diff
@@ -0,0 +1,252 @@
+import io.pravega.auth.TokenExpiredException;
+import io.pravega.client.connection.impl.ConnectionPool;
+import io.pravega.client.connection.impl.RawClient;
+import io.pravega.client.control.impl.Controller;
+import io.pravega.client.security.auth.DelegationTokenProvider;
+import io.pravega.client.segment.impl.NoSuchSegmentException;
+import io.pravega.client.segment.impl.Segment;
+import io.pravega.client.segment.impl.SegmentSealedException;
+import io.pravega.client.stream.EventWriterConfig;
+import io.pravega.client.stream.Serializer;
+import io.pravega.common.util.RetriesExhaustedException;
+import io.pravega.shared.protocol.netty.ConnectionFailedException;
+import io.pravega.shared.protocol.netty.Reply;
+import io.pravega.shared.protocol.netty.WireCommandType;
+import io.pravega.shared.protocol.netty.WireCommands;
+import io.pravega.shared.protocol.netty.WireCommands.AppendSetup;
+import io.pravega.shared.protocol.netty.WireCommands.AuthTokenCheckFailed;
+import io.pravega.shared.protocol.netty.WireCommands.ConditionalBlockEnd;
+import io.pravega.shared.protocol.netty.WireCommands.ConditionalCheckFailed;
+import io.pravega.shared.protocol.netty.WireCommands.CreateTransientSegment;
+import io.pravega.shared.protocol.netty.WireCommands.DataAppended;
+import io.pravega.shared.protocol.netty.WireCommands.InvalidEventNumber;
+import io.pravega.shared.protocol.netty.WireCommands.MergeSegments;
+import io.pravega.shared.protocol.netty.WireCommands.OperationUnsupported;
+import io.pravega.shared.protocol.netty.WireCommands.SegmentAlreadyExists;
+import io.pravega.shared.protocol.netty.WireCommands.SegmentCreated;
+import io.pravega.shared.protocol.netty.WireCommands.SegmentIsSealed;
+import io.pravega.shared.protocol.netty.WireCommands.SegmentIsTruncated;
+import io.pravega.shared.protocol.netty.WireCommands.SegmentsMerged;
+import io.pravega.shared.protocol.netty.WireCommands.SetupAppend;
+import io.pravega.shared.protocol.netty.WireCommands.WrongHost;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import lombok.Cleanup;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import lombok.extern.slf4j.Slf4j;
+
+import static io.pravega.common.concurrent.Futures.getThrowingException;
+
+@RequiredArgsConstructor
+@Slf4j
+public class LargeEventWriter {
+
+    private static final int WRITE_SIZE = Serializer.MAX_EVENT_SIZE;
+    private final UUID writerId;
+    private final Controller controller;
+    private final ConnectionPool connectionPool;
+
+    public void writeLargeEvent(Segment segment, List<ByteBuffer> events, DelegationTokenProvider tokenProvider,
+            EventWriterConfig config) throws NoSuchSegmentException, AuthenticationException, SegmentSealedException {
+        List<ByteBuf> payloads = createBufs(events);
+        int attempts = 1 + Math.max(0, config.getRetryAttempts());
```

Comment:
`EventWriterConfig` ensures that the retry attempts is non-negative so I think you can omit the `Math.max` call.

Reply:
The builder enforces it, but because it is serializable, it technically can be non-zero if the bits of the serialized form have been modified.